### PR TITLE
refactor(core): unwrap sync ops in rust

### PR DESCRIPTION
**Pull request by nayeemrmn**
Blocked on #14734.
Assists with #15337.
Supersedes #14811.

### CODE REVIEW

**Comment by nayeemrmn** [comment]
Turns out the error builders need to be registered separately per-realm, so this depends on the architecture in #14734 (@andreubotella )

File: core/error.rs [hunk 1]
```diff
@@ -91,6 +92,24 @@ pub fn get_custom_error_class(error: &Error) -> Option<&'static str> {
...
+) -> v8::Local<'a, v8::Value> {
+  let cb = JsRuntime::state(scope)
+    .borrow()
+    .js_build_custom_error_cb
+    .clone()
+    .expect("Custom error builder must be set");
+  let cb = cb.open(scope);
+  let this = v8::undefined(scope).into();
+  let class = v8::String::new(scope, get_class(error)).unwrap();
+  let message = v8::String::new(scope, &error.to_string()).unwrap();
```
Comment (aapoalas) [inline core/error.rs:108]: Errors can also have the `code` parameter so I guess that should be created here as well (or undefined if not used, reuse `this`) and passed to the call.

Comment (nayeemrmn) [reply core/error.rs:108]: Added to `to_v8_error()`

**Comment by nayeemrmn** [comment]
Accidentally removed changes from the PR 😅 will probably have to open a new one.

**Pull request merged**